src/components/DataSheetGrid.tsx

```diff
@@ -1631,6 +1632,16 @@ export const DataSheetGrid = React.memo(
         [headerRowHeight, rowHeight]
       )
 
+      const itemKey = useCallback(
+        (index: number, { data }: ListItemData<T>) => {
+          const row = data[index - 1] as Record<any, any>
+          return index === 0 || !rowKey || !(rowKey in row)
+            ? index
+            : row[rowKey]
```

Comment:
This should be revised to support a function or a string as `rowKey`.
We also probably need to check that row is an object (since it is now a `T`)